packages/ozone/src/tag-service/language-tagger.ts

```diff
@@ -5,46 +5,29 @@ import {
   AppBskyGraphList,
 } from '@atproto/api'
 
-import { ModerationService } from '.'
-import { ModSubject } from './subject'
-import { ModerationSubjectStatusRow } from './types'
 import { langLogger as log } from '../logger'
-import { code3ToCode2 } from './lang-data'
+import { code3ToCode2 } from './language-data'
+import { ContentTagger } from './content-tagger'
 
-export class ModerationLangService {
-  constructor(private moderationService: ModerationService) {}
+export class LanguageTagger extends ContentTagger {
+  tagPrefix = 'lang:'
 
-  async tagSubjectWithLang({
-    subject,
-    subjectStatus,
-    createdBy,
-  }: {
-    subject: ModSubject
-    createdBy: string
-    subjectStatus: ModerationSubjectStatusRow | null
-  }) {
-    if (
-      subjectStatus &&
-      !subjectStatus.tags?.find((tag) => tag.includes('lang:'))
-    ) {
-      try {
-        const recordLangs = await this.getRecordLang({
-          subject,
-        })
-        await this.moderationService.logEvent({
-          event: {
-            $type: 'tools.ozone.moderation.defs#modEventTag',
-            add: recordLangs
-              ? recordLangs.map((lang) => `lang:${lang}`)
-              : ['lang:und'],
-            remove: [],
-          },
-          subject,
-          createdBy,
-        })
-      } catch (err) {
-        log.error({ subject, err }, 'Error getting record langs')
-      }
+  isApplicable(): boolean {
+    return Boolean(
+      this.subjectStatus &&
+        !this.subjectStatus.tags?.find((tag) => tag.includes(this.tagPrefix)),
```

Comment:
Just a little more specific!
```suggestion
        !this.subjectStatus.tags?.some((tag) => tag.startsWith(this.tagPrefix)),
```

Reply:
thanks, didn't like having to change this in 2 different places so refactored the class abstract class a bit more to move this logic in one place.